src/zenml/orchestrators/step_runner.py

```diff
@@ -567,14 +574,25 @@ def _store_output_artifacts(
 
             uri = output_artifact_uris[output_name]
             materializer = materializer_class(uri)
+            has_custom_name = output_has_custom_name[output_name]
+
+            # Override the artifact name if it is not a custom name.
+            if has_custom_name:
+                artifact_name = output_name
```

Comment:
For the reasons I outlined in PRD, I would not go with this but rather have a `f"{pipeline_name}::{step_name}::{output_name}"` pattern all over the place.

Reply:
@schustmi and I discussed this before and decided to not do that. 

The main reason is that you might want to create new versions of the same artifact from different pipelines or steps, or you might want to rename your pipeline or steps at some point which would break the lineage if custom names also need to follow this prefix.

Also, if users want to follow this naming convention, they can easily do so themselves, so why force them